jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/PropertiesByNameRule.java

```diff
@@ -0,0 +1,371 @@
+/**
+ * Copyright ¬© 2010-2014 Nokia
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jsonschema2pojo.rules;
+
+import java.util.Iterator;
+import java.util.Map;
+
+import org.jsonschema2pojo.Schema;
+import org.jsonschema2pojo.util.LanguageFeatures;
+import org.jsonschema2pojo.util.Models;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.sun.codemodel.JBlock;
+import com.sun.codemodel.JCodeModel;
+import com.sun.codemodel.JConditional;
+import com.sun.codemodel.JDefinedClass;
+import com.sun.codemodel.JExpr;
+import com.sun.codemodel.JExpression;
+import com.sun.codemodel.JFieldRef;
+import com.sun.codemodel.JGenerifiable;
+import com.sun.codemodel.JInvocation;
+import com.sun.codemodel.JMethod;
+import com.sun.codemodel.JOp;
+import com.sun.codemodel.JSwitch;
+import com.sun.codemodel.JType;
+import com.sun.codemodel.JTypeVar;
+import com.sun.codemodel.JVar;
+import com.sun.codemodel.JClass;
+
+import static com.sun.codemodel.JExpr.*;
+import static com.sun.codemodel.JMod.*;
+import static org.jsonschema2pojo.util.LanguageFeatures.*;
+
+public class PropertiesByNameRule implements Rule<JDefinedClass, JDefinedClass> {
```

Comment:
This rule should be called DynamicPropertiesRule, to align naming with the flag.